### PR TITLE
refactor: move API schemas folder structure

### LINKED ISSUE #4868: [REFACTOR] Improve argilla-server folder structure
We should improve folder structure on `argilla-server` with the following:

```
/api
  /handlers
    /v1
    /v2
  /policies
    /v1
    /v2
  /schemas
    /v1
    /v2
/contexts
/models
...
```

So the idea is to have everything related with the API inside an `api` folder. Inside that folder we should have `handlers`,  `policies` and `schemas`.

For stuff not directly related with the versioned API we can have folders outside of `api`, like `/context` and `/models`.

Because these are changes that are not really complex to do and in order to avoid conflicts with current changes I would do this issue over branch `feat/v2.0.0`.

cc @frascuchon

### PR DESCRIPTION
**Pull request by jfcalvo**
# Description

This PR include changes moving API schemas to follow this structure:
```
/api
  /schemas
    /v1
      datasets.py
      records.py
      ...
```

The behavior has not been changed.

Refs #4868

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)
- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [x] Test should be passing as before.

**Checklist**

- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla/tests/unit/client/sdk/models/test_workspaces.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 #  limitations under the License.
 
 from argilla.client.sdk.workspaces.models import WorkspaceModel as ClientSchema
-from argilla_server.schemas.v1.workspaces import Workspace as ServerSchema
+from argilla_server.api.schemas.v1.workspaces import Workspace as ServerSchema
```
Comment (frascuchon) [inline argilla/tests/unit/client/sdk/models/test_workspaces.py]: I think we shouldn't touch the "old" argilla client. We can just fix the server version to 1.29.0 and the code remains compatible.

File: argilla/pyproject.toml [hunk 2]
```diff
@@ -48,14 +48,14 @@ dynamic = ["version"]
...
-    "argilla-server ~= 1.29.0"
+    # "argilla-server ~= 1.29.0"
 ]
 server-postgresql = [
-    # "argilla-server[postgresql]"
-    # Comment previos line and uncomment the next line when we have a release
-    "argilla-server[postgresql] ~= 1.29.0"
+    "argilla-server[postgresql]",
+    # Comment previous line and uncomment the next line when we have a release
+    # "argilla-server[postgresql] ~= 1.29.0"
```
Comment (frascuchon) [inline argilla/pyproject.toml]: The current argilla-server won't be compatible. We should let this fixed, even for development.

File: argilla-server/src/argilla_server/models/database.py [hunk 3]
```diff
@@ -24,6 +24,7 @@
 from sqlalchemy.ext.mutable import MutableDict, MutableList
 from sqlalchemy.orm import Mapped, mapped_column, relationship
 
+from argilla_server.api.schemas.v1.questions import QuestionSettings
```
Comment (frascuchon) [inline argilla-server/src/argilla_server/models/database.py:27]: Ideally, we should avoid schema imports in core model modules.

**Review by frascuchon — CHANGES_REQUESTED** [review]
Same comment as handlers PR. We should remove all the changes for the `argilla` project.

File: argilla-server/src/argilla_server/models/database.py (hunk 3 above)
Comment (jfcalvo) [reply argilla-server/src/argilla_server/models/database.py:27]: I agree. That one is needed by the infamous `Question.parsed_settings` function.

**Review by frascuchon — APPROVED** [review]

**Pull request merged**